### ISSUE TITLE
show the actual tags that will be applied

### ISSUE THREAD

**Issue by thibaultmol** [OPENER-next/OpenStop#255]
### Idea

It might be good for people who actually know whatt osm is if there was an option in the settings to show the actual tag(s) that will be applied depending on your answer.

**Comment by rugk** [comment]
IMHO rather an advanced feature and I like how StreetComplete solves it to only show this in the "revert dialog", which would be a new feature for this app, too.

**Comment by thibaultmol** [comment]
Is it that advanced, though? The app already has to apply the tags anyway, it could just show them? that's how Mapcomplete does it:
![image](https://github.com/OPENER-next/OpenStop/assets/1010226/6db2cd33-b490-4de1-a1fa-6e0040a870be)

It even has a link to the wiki page for each key/tag but that's also easy to do because it's just a url template:
`https://wiki.openstreetmap.org/wiki/Tag:bench%3Dyes`

**Comment by rugk** [comment]
Yeah, StreetComplete has in the past refrained from that, because it would just confuse a typical user (as they do not even [have to] know OSM or so), so depending on your target audience, I'd also argue this is similar here.
Also on smartphones there is limited space and if you want to modify multiple tags it gets more etc. etc. SC has reasons not to show it (could not find a fitting issue in their repo right now though).

**Comment by thibaultmol** [comment]
I somewhat get it. But on other hand I'm like: it's just one or two lines of text.

Here's how mapcomplete does it on mobile (and that's even with the mode on that shows all questions at once which isn't the default setting for mapcomplete 
![Screenshot_20240331-085931](https://github.com/OPENER-next/OpenStop/assets/1010226/69e06cd5-ffcf-47a3-86bf-5102ef56f871)

**Comment by rugk** [comment]
Yeah, sure, but I TBH Mapcomplete is not really an example of great UX, IMHO. Anyway, sure, do it by all means, maybe hide it behind an `(i)` icon or so, there are likely more ideas or so, I just wanted to bring in some arguments of why this may not be the most important idea or one which also had some disadvantages.

**Comment by 1231231231231231231231231** [comment]
The limited screen space is a valid concern, also because people shouldn't be overwhelmed when this is aimed at an even more general audience than StreetComplete already is.

On the other hand, one of my main annoyances in StreetComplete (second to map rotation constantly triggering when trying to zoom) is that I can't easily see what tags it sets. This is better nowadays with the undo feature showing the tags at least, but it's a workaround more than a solution because you have to pick one of the solutions and then go back to see what you've actually done (and make sure to have auto-upload disabled).

To preserve screen space and visual clutter, could it be something you can fold out? Or if you long press the answer box or options, it will expand or so? It can tell users this is possible in the settings screen for example, then advanced users can find it and new users are not bothered with it

**Comment by 7h30n3** [comment]
This seems quite an controversal topic, but we want to elaborate on this. 
We have some possible solutions in mind, but need feedback from you on which way to go.

The first idea was to just show the tag below the corresponding entry on the summery screen. However not every answer shown there gets acutally written in the changeset. For example, if you answer "Is there a kerb?" with "Yes" (kerb=yes) and later "Is the kerb raised?" also with "Yes" (kerb=raised), only the last tag will be written dispite both answers are shown on the summery screen.
Of course, the "unwritten" tag can just be grayed out.

The second idea was to make it more sophisticated. To get a real benefit from this you not only want to know what new tags will be added, but also what are the current tags. So you can make a qualified descision, if it's all correct.
This means a possible approach could be to show all tags and only the tags. Each tag will be in one of 4 separate "boxes" (unmodified, added, deleted, modified value), to make them easily and quickly distinquishable.

We thought about adding the switch to toggle between the default and the expert view on the summary sheet and not in the settings menu. The toogle would not be persistent, so you always have to click on that button to switch views, because we think the feature/ expert view should only be necessary in rare occasions.

What's your opionion on that?

**Comment by thibaultmol** [comment]
I think the 1st option is fine. if people want more, they can open osm directly

**Comment by 1231231231231231231231231** [comment]
> So you can make a qualified descision, if it's all correct.

This is why the second solution would have my preference.

Instead of needing to use "boxes", a format like what `git diff` shows seems fairly standardised (removed lines in red with a `-` prefixed, added lines in green with a `+` prefixed). Looking into how red-green colour-blind people handle that, since that is the majority of all color-blind people (8% of men), people suggest either shifting the red a bit to orange and the green a bit to cyan, or creating a difference in shading. I found a red-green color-blind person saying that GitHub's red/green diffs work for them so I checked and the green is a lot more saturated than the red. Perhaps that creates a visible-to-everyone shading effect; I'm not sure

> The toogle would not be persistent, so you always have to click on that button to switch views, because we think the feature/ expert view should only be necessary in rare occasions

It not not being necessary sounds like a good principle

If that view toggle is presented every time you make a change, that also makes it more visible to everyone and invites people to look at how OSM works with the underlying tags. I would not be opposed to that at all: simple by default, expert view a tap away. Helps people learn how to do the same in other editors as well

I don't think I'd mind doing a single extra tap, even if I would do it most times, especially as compared to StreetComplete where you need to do a number of taps across the screen to apply a proposed change, then get to the undo menu and view the changed tags

**Comment by rugk** [comment]
Yeah v1 would be fine for me. If you want v2 then maybe an "expansion" button (cf [this older material design](https://m1.material.io/components/expansion-panels.html); [also in Angular](https://v5.material.angular.io/components/expansion/overview)) at the bottom or so could help? Or, as you currently kinda have, [a bottom sheet](https://m3.material.io/components/bottom-sheets/overview), but you can drag it out further to see more details… ([more about that here](https://m3.material.io/components/bottom-sheets/guidelines#2437a6ca-9a99-40eb-8c71-fe7093200521))

I agree some colored diff with red/green for added/removed would be good. For related art, maybe one should look into how the OSM editor Everydoor shows it.
But in any case, one should not be overwhelmed with that, so keep it as simple as possible (or hidden as indicated) and only offer it as some insight if it is wanted.

These are just some visual ideas.
